src/type-generator.ts

```diff
@@ -111,8 +136,32 @@ export class TypeGenerator {
     });
   }
 
+  private buildEnumType(enumType: pgStructure.Type) {
+    const types = (enumType as any).values.map((t: string) => `'${t}'`).join(
+      ' | ',
+    );
+
+    return { name: this.getEnumTypeName(enumType), type: types };
+  }
+
+  private buildEnumTypes(db: pgStructure.Db) {
+    const enumTypes = [];
+    for (const t of db.types) {
+      if (t.category === 'E') {
+        enumTypes.push(this.buildEnumType(t));
+      }
+    }
+
+    return enumTypes;
+  }
+
   private generateType() {
     const dbTypings = this.buildDBType(this.db!);
+    const enumTypes = this.buildEnumTypes(this.db!);
+
+    const enumTypings = enumTypes.map((t) =>
+      `export type ${t.name} = ${t.type}`
```

Comment:
I see you're mapping them to unions instead of enums on ts side, mind leaving a comment on why?

Reply:
The primary motivation is typecheking and code completion. Also, the ability to use literal strings as values without importing and using the enum